src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/PublicSurfaceTests.cs

```diff
@@ -0,0 +1,106 @@
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CodeActionParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CodeActionResult",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CompletionContext",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CompletionItem",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CompletionItemKind",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CompletionList",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CompletionParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CompletionTriggerKind",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.CustomProtocolNames",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.Diagnostic",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.DiagnosticSeverity",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.DidChangeTextDocumentParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.DidOpenTextDocumentParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.InitialFixupParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.ParameterInformation",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.Position",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.PublishDiagnosticsParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.PublishExpressionTypeParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.PublishTokensParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.Range",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.SignatureHelp",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.SignatureHelpContext",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.SignatureHelpParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.SignatureHelpTriggerKind",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.SignatureInformation",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.TextDocumentContentChangeEvent",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.TextDocumentIdentifier",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.TextDocumentItem",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.TextDocumentNames",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.TextDocumentPositionParams",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.TextEdit",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.VersionedTextDocumentIdentifier",
+                "Microsoft.PowerFx.LanguageServerProtocol.Protocol.WorkspaceEdit",
```

Comment:
We never really looked at the public surface here, but I think none of these Protocol classes need to be public, they're just POCOs for serialization.

Reply:
agreed - I was just capturing the current state. 
Note that the *wire protocol* is public, but the model classes we use here don't need to be.  (in fact, I'm hoping there's some existing LSP nuget that defines these already so that we don't have to define).

Reply:
#660